Extend examples for AtomicBool::{swap, compare_and_swap}

jooert opened the pull request:
Closes #26169.

r? @steveklabnik

alexcrichton commented on src/libcore/atomic.rs:
```diff
@@ -285,7 +286,8 @@ impl AtomicBool {
     /// let some_bool = AtomicBool::new(true);
     ///
-    /// let value = some_bool.store(false, Ordering::Relaxed);
+    /// assert_eq!(true, some_bool.compare_and_swap(false, false, Ordering::Relaxed));
```
Could this also perhaps show off an example of a successful compare-and-swap? Also, our assertions are typically `assert_eq!(actual, expected)` instead of the other way around I believe.

jooert replied:
I prefer `assert_eq!(actual, expected)` as well, but all other examples for `AtomicBool` used `assert_eq!(expected, actual)` and I wanted to stay consistent. Should I swap the arguments of all other asserts in this file as well?

alexcrichton replied:
Sure!

jooert commented:
Added successful `compare_and_swap` examples, changed order of arguments when calling `assert_eq!` and added some additional assertions in other examples.

alexcrichton commented:
@bors: r+ 5c9f069ece7a2d1bdbf79f085fe96de0d4033f0f 

Thanks!

bors commented:
:hourglass: Testing commit 5c9f069 with merge 7e9a8e3...

bors commented:
:sunny: Test successful - [auto-linux-32-nopt-t](http://buildbot.rust-lang.org/builders/auto-linux-32-nopt-t/builds/5307), [auto-linux-32-opt](http://buildbot.rust-lang.org/builders/auto-linux-32-opt/builds/5318), [auto-linux-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-linux-64-nopt-t/builds/5270), [auto-linux-64-opt](http://buildbot.rust-lang.org/builders/auto-linux-64-opt/builds/5296), [auto-linux-64-x-android-t](http://buildbot.rust-lang.org/builders/auto-linux-64-x-android-t/builds/5265), [auto-mac-32-opt](http://buildbot.rust-lang.org/builders/auto-mac-32-opt/builds/5341), [auto-mac-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-mac-64-nopt-t/builds/5333), [auto-mac-64-opt](http://buildbot.rust-lang.org/builders/auto-mac-64-opt/builds/5302), [auto-win-gnu-32-nopt-t](http://buildbot.rust-lang.org/builders/auto-win-gnu-32-nopt-t/builds/283), [auto-win-gnu-32-opt](http://buildbot.rust-lang.org/builders/auto-win-gnu-32-opt/builds/283), [auto-win-gnu-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-win-gnu-64-nopt-t/builds/282), [auto-win-gnu-64-opt](http://buildbot.rust-lang.org/builders/auto-win-gnu-64-opt/builds/282)
